### PR TITLE
Sha256

### PR DESCRIPTION
**Pull request by Duncaen**

### CODE REVIEW

File: lib/transaction_files.c [hunk 1]
```diff
@@ -75,6 +75,7 @@ itemhash(const char *file)
 
 	assert(file);
 
+	/* XXX: runtime error: left shift of negative value -1581911230 */
```
Comment (xtraeme) [inline lib/transaction_files.c]: maybe bump ITHSIZE ?

File: lib/package_config_files.c [hunk 2]
```diff
@@ -232,15 +231,11 @@ xbps_entry_install_conf_file(struct xbps_handle *xhp,
 			rv = 1;
 			break;
 		}
-		free(sha256_cur);
-		sha256_cur = NULL;
 	}
 
 out:
 	if (sha256_orig)
 		free(sha256_orig);
```
[the remark this answers is not in the document]
Comment (xtraeme) [reply lib/package_config_files.c]: sha256_orig I meant

File: include/xbps.h.in [hunk 3]
```diff
@@ -1882,7 +1886,7 @@ unsigned char *xbps_file_hash_raw(const char *file);
  * @return 0 if \a file and \a sha256 have the same hash, ERANGE
  * if it differs, or any other errno value on error.
  */
-int xbps_file_hash_check(const char *file, const char *sha256);
+int xbps_file_sha256_check(const char *file, const char *sha256);
```
[the remark this answers is not in the document]
Comment (xtraeme) [reply include/xbps.h.in]: You can use the updated descriptions from my patch: https://patch-diff.githubusercontent.com/raw/void-linux/xbps/pull/223.diff

File: lib/transaction_files.c (hunk 1 above)
Comment (xtraeme) [reply lib/transaction_files.c]: this is resolved, we'll use uthash soon :-)

**Comment by xtraeme** [comment]
this is great! please merge

**Comment by Duncaen** [comment]
Merged :+1:

**Pull request closed without merging**